MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -345,19 +354,81 @@ - (void)onProcessingLog:(id<MSLog>)log withPriority:(MSPriority)priority {
     NSData *serializedLog = [NSKeyedArchiver archivedDataWithRootObject:log];
     if (serializedLog && (serializedLog.length > 0)) {
 
-      // Our arrays contain a max of 20 items, so our maxIndex == 19.
-      int maxIndex = ms_crashes_log_buffer_size - 1;
-      if (self.bufferIndex[@(priority)].integerValue > maxIndex) {
+      NSNumber *oldestTimestamp;
+      NSNumberFormatter *timestampFormatter = [[NSNumberFormatter alloc] init];
+      timestampFormatter.numberStyle = NSNumberFormatterDecimalStyle;
+      long indexToDelete = 0;
+      for (auto it = msCrashesLogBuffer[priority].begin(), end = msCrashesLogBuffer[priority].end(); it != end; ++it) {
```

Comment:
Not doing this because using the iterator is very straight forward and less prone to errors than using indices. Using an iterator now might be a tiny bit slower than the indices but we don't need to optimize for speed more for now.